src/Compilers/CSharp/Portable/Binder/UsingStatementBinder.cs

```diff
@@ -79,18 +79,29 @@ internal override BoundStatement BindUsingStatementParts(DiagnosticBag diagnosti
                 iDisposableConversion = originalBinder.Conversions.ClassifyImplicitConversionFromExpression(expressionOpt, iDisposable, ref useSiteDiagnostics);
                 diagnostics.Add(expressionSyntax, useSiteDiagnostics);
 
-                if (!iDisposableConversion.IsImplicit)
+                TypeSymbol expressionType = expressionOpt.Type;
+                
+                if (!iDisposableConversion.IsImplicit && (object)disposeMethod is null)
                 {
-                    TypeSymbol expressionType = expressionOpt.Type;
-                    if ((object)expressionType == null || !expressionType.IsErrorType())
+                    if (!(expressionType is null))
                     {
-                        Error(diagnostics, ErrorCode.ERR_NoConvToIDisp, expressionSyntax, expressionOpt.Display);
+                        disposeMethod = TryFindDisposePatternMethod(expressionType, diagnostics);
                     }
-                    hasErrors = true;
+                    if (disposeMethod is null)
+                    {
+                        if ((object)expressionType == null || !expressionType.IsErrorType())
+                        {
+                            Error(diagnostics, ErrorCode.ERR_NoConvToIDisp, expressionSyntax, expressionOpt.Display);
+                        } else
```

Comment:
The `else` here looks wrong. `hasErrors` should always be set to `true`. The `if (disposeMethod is null)` should be enough to ensure we're in an error case.